test/functional/test-intersection-observer-polyfill.js

```diff
@@ -0,0 +1,397 @@
+    });
+
+    it('should NOT contain value less than 0 or greater than 1', () => {
+      const io1 = () => {
+        new IntersectionObserverPolyfill(() => {}, {
+          threshold: [0, -2, 0.3, 1, 0.02],
+        });
+      };
+      const io2 = () => {
+        new IntersectionObserverPolyfill(() => {}, {
+          threshold: [0, 1.1, 0.3, 1, 0.02],
+        });
+      };
+      expect(io1).to.throw(
+          'Threshold should only contain value range from 0 to 1');
+      expect(io2).to.throw(
+          'Threshold should only contain value range from 0 to 1');
+    });
+
+    it('getThresholdSlot function', () => {
+      let threshold = DEFAULT_THRESHOLD;
+      expect(getThresholdSlot(threshold, 0)).to.equal(0);
+      expect(getThresholdSlot(threshold, 0.67)).to.equal(14);
+      expect(getThresholdSlot(threshold, 0.65)).to.equal(14);
+      expect(getThresholdSlot(threshold, 1)).to.equal(21);
+      threshold = [0.01, 0.25, 0.5, 1];
+      expect(getThresholdSlot(threshold, 0)).to.equal(0);
+      threshold = [0, 0.25, 0.5, 1];
+      expect(getThresholdSlot(threshold, 0)).to.equal(0);
+      expect(getThresholdSlot(threshold, 0.1)).to.equal(1);
+      expect(getThresholdSlot(threshold, 0.4)).to.equal(2);
+      expect(getThresholdSlot(threshold, 0.25)).to.equal(2);
+      expect(getThresholdSlot(threshold, 0.5)).to.equal(3);
+      expect(getThresholdSlot(threshold, 1)).to.equal(4);
+      threshold = [0.5];
+      expect(getThresholdSlot(threshold, 1)).to.equal(1);
+    });
+
+  });
+
+  describe('tick function', () => {
+    let element;
+    let callbackSpy;
+
+    let io;
+    beforeEach(() => {
+      callbackSpy = sandbox.spy();
+      io = new IntersectionObserverPolyfill(callbackSpy);
+      element = {
+        isBuilt: () => {return true;},
+        getOwnerLayoutBox: () => {return null;},
+      };
+    });
+
+    afterEach(() => {
+      element = null;
+    });
+
+
+    describe('w/o container should get IntersectionChangeEntry when', () => {
+      it('intersect correctly base', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 50, 150, 200);
+        };
+        io.observe(element);
+        const rootBounds = layoutRectLtwh(0, 100, 100, 100);
+        io.tick(rootBounds);
+        const change = {
+          time: 100,
+          rootBounds: DomRectLtwh(0, 0, 100, 100),
+          boundingClientRect: DomRectLtwh(50, -50, 150, 200),
+          intersectionRect: DomRectLtwh(50, 0, 50, 100),
+          intersectionRatio: 1 / 6,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('intersects on the edge', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 200, 150, 200);
+        };
+        const rootBounds = layoutRectLtwh(0, 100, 100, 100);
+        io.observe(element);
+        io.prevThresholdSlot_ = -1;
```

Comment:
I really don't want to  change the private field here. BUT a changing to this private field makes testing much easier, LOL 😢